editor/plugins/tiles/tile_set_atlas_source_editor.cpp

```diff
@@ -1100,7 +1101,7 @@ void TileSetAtlasSourceEditor::_tile_atlas_control_gui_input(const Ref<InputEven
 				}
 			} else if (drag_type >= DRAG_TYPE_RESIZE_TOP_LEFT && drag_type <= DRAG_TYPE_RESIZE_LEFT) {
 				// Resizing a tile.
-				new_base_tiles_coords = new_base_tiles_coords.max(Vector2i(-1, -1)).min(grid_size);
+				new_base_tiles_coords = tile_atlas_view->get_atlas_tile_coords_at_pos(tile_atlas_control->get_local_mouse_position(), TileAtlasView::CLAMP_MARGIN);
```

Comment:
If this is the only place where this is needed, I believe modifying so much code is not really needed.
Instead,  I think you should use the unclamped version and clamp it manually like how it was done before. Basically you would just replace this line by:
```
new_base_tiles_coords = tile_atlas_view->get_atlas_tile_coords_at_pos(tile_atlas_control->get_local_mouse_position()).max(Vector2i(-1, -1)).min(grid_size);
```